partiql-ast/src/main/java/org/partiql/ast/v1/DataType.java

```diff
@@ -0,0 +1,318 @@
+package org.partiql.ast.v1;
+
+public class DataType implements Enum {
+    public static final int UNKNOWN = 0;
+    public static final int NULL = 1;
+    public static final int MISSING = 2;
+    public static final int BOOL = 3;
+    public static final int BOOLEAN = 4;
+    public static final int TINYINT = 5; // TODO not defined in parser yet
+    public static final int SMALLINT = 6;
+    public static final int INTEGER2 = 7;
+    public static final int INT2 = 8;
+    public static final int INTEGER = 9;
+    public static final int INT = 10;
+    public static final int INTEGER4 = 11;
+    public static final int INT4 = 12;
+    public static final int INTEGER8 = 13;
+    public static final int INT8 = 14;
+    public static final int BIGINT = 15;
+    public static final int REAL = 16;
+    public static final int DOUBLE_PRECISION = 17;
+    public static final int FLOAT = 18;
+    public static final int DECIMAL = 19;
+    public static final int DEC = 20;
+    public static final int NUMERIC = 21;
+    public static final int BIT = 22; // TODO not defined in parser yet
+    public static final int BIT_VARYING = 23; // TODO not defined in parser yet
+    public static final int CHAR = 24;
+    public static final int CHARACTER = 25;
+    public static final int VARCHAR = 26;
+    public static final int CHARACTER_VARYING = 27;
```

Comment:
Let's group things together based on SQL standard definitions. 

For example,
```java
// <character string type>
public static final int CHARACTER = 24;
public static final int CHAR = 25;
public static final int CHARACTER_VARYING = 26;
public static final int CHAR_VARYING = 27;
public static final int VARCHAR = 28;
public static final int CHARACTER_LARGE_OBJECT = 29;
public static final int CHAR_LARGE_OBJECT = 30;
public static final int CLOB = 31; 
```

Which would make implementations of some function `isCharacterString()` easy/efficient to answer aka code in range.

This will also help us cover all SQL things first, then we can have a separate codespace for extended type syntax. I would skip `<national character string type>`..

Reply:
Good idea. I wasn't too sure on the best way to group the constants. Will try to follow SQL's grouping/ordering.

Will also skip `<national character string type>`